src/Altinn.App.Core/Internal/Events/UnhandledEventHandler.cs

```diff
@@ -0,0 +1,21 @@
+﻿using Altinn.App.Core.Features;
+using Altinn.App.Core.Models;
+using System.Text.Json;
+
+namespace Altinn.App.Core.Internal.Events
+{
+    /// <summary>
+    /// Implementation used to handled events that could not bee resolved and matched on type.
+    /// </summary>
+    public class UnhandledEventHandler : IEventHandler
+    {
+        /// <inheritdoc/>
+        public string EventType => "app.events.unhandled";
+
+        /// <inheritdoc/>
+        public Task<bool> ProcessEvent(CloudEvent cloudEvent)
+        {
+            throw new NotImplementedException($"Received unhandled event {cloudEvent.Type} with the following data: {JsonSerializer.Serialize(cloudEvent)}");
```

Comment:
Will this break the code if we introduce a new event that old applications dont know about?

Reply:
Not sure if I follow. This code will only be invoked if a given application receives an event it hasn't subscribed to.

Reply:
More likely that I'm not following. The application subscribes to a defined list of events and only them? If so it should be ok to throw a NotImplementedException.

Reply:
Yep! This is basically saying to the Event System that "I haven't implemented support for the event you just sent me!". I case very unlikely to happen though, but at least we then have it explicit if it should happen.